File: shared/domain/src/commonMain/kotlin/com/zyntasolutions/zyntapos/domain/model/MasterProduct.kt
package com.zyntasolutions.zyntapos.domain.model

import kotlinx.datetime.Instant

/**
 * A centrally-managed product template shared across all stores.
 *
 * Master products live in the global catalog (admin-panel-only writes).
 * POS devices receive them via sync pull as read-only data.
 * Stores can override [basePrice] and [costPrice] locally via [StoreProductOverride].
 *
 * @property id Unique identifier (UUID v4).
 * @property sku Global stock-keeping unit code. Unique across all stores.
 * @property barcode EAN-13 / Code128 barcode value. Unique globally.
 * @property name Display name shown on receipts, POS grid, and reports.
 * @property description Optional rich-text description.
 * @property basePrice Default selling price. Stores may override this locally.
 * @property costPrice Default purchase cost. Never shown to customers.
 * @property categoryId FK to [Category].
 * @property unitId FK to [UnitOfMeasure] (e.g., "pcs", "kg").
 * @property taxGroupId FK to [TaxGroup]. Null means no tax applied.
 * @property imageUrl Remote or local URI for the product image.
 * @property isActive If false the product is hidden from the catalog.
 * @property createdAt UTC timestamp of record creation.
 * @property updatedAt UTC timestamp of the last modification.
 */
data class MasterProduct(
    val id: String,
    val sku: String? = null,
    val barcode: String? = null,
    val name: String,
    val description: String? = null,
    val basePrice: Double,
    val costPrice: Double = 0.0,
    val categoryId: String? = null,
    val unitId: String? = null,
    val taxGroupId: String? = null,
    val imageUrl: String? = null,
    val isActive: Boolean = true,
    val createdAt: Instant,
    val updatedAt: Instant,
)

File: admin-panel/src/types/master-product.ts
export interface MasterProduct {
  id: string;
  sku: string | null;
  barcode: string | null;
  name: string;
  description: string | null;
  base_price: number;
  cost_price: number;
  category_id: string | null;
  unit_id: string | null;
  tax_group_id: string | null;
  image_url: string | null;
  is_active: boolean;
  created_at: number;
  updated_at: number;
  store_count: number;
}

export interface StoreProductAssignment {
  store_id: string;
  store_name: string;
  local_price: number | null;
  local_cost_price: number | null;
  local_stock_qty: number;
  min_stock_qty: number;
  is_active: boolean;
}

export interface CreateMasterProductRequest {
  sku?: string;
  barcode?: string;
  name: string;
  description?: string;
  base_price: number;
  cost_price?: number;
  category_id?: string;
  unit_id?: string;
  tax_group_id?: string;
  image_url?: string;
}

export interface UpdateMasterProductRequest {
  sku?: string;
  barcode?: string;
  name?: string;
  description?: string;
  base_price?: number;
  cost_price?: number;
  category_id?: string;
  unit_id?: string;
  tax_group_id?: string;
  image_url?: string;
  is_active?: boolean;
}

export interface AssignToStoreRequest {
  local_price?: number;
  local_cost_price?: number;
  local_stock_qty?: number;
  min_stock_qty?: number;
}

export interface BulkAssignRequest {
  store_ids: string[];
  local_price?: number;
  local_cost_price?: number;
}

export interface MasterProductFilter {
  page?: number;
  size?: number;
  search?: string;
}
